flang/lib/Semantics/check-call.cpp

```diff
@@ -1145,6 +1145,37 @@ static void CheckExplicitDataArg(const characteristics::DummyDataObject &dummy,
           dummyName, toStr(dummyDataAttr), toStr(actualDataAttr));
     }
   }
+  // Emit an error message if an actual argument passed to a host intrinsic is
+  // on the device.
```

Comment:
What about host data being used in an intrinsic procedure reference on the device?  Is that caught by other means already?